app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -68,14 +68,29 @@ def reset!
         DataCiteEndpoint.reset!
         Settings.switch!
         switch_host!(nil)
+        disable_tenant_cache
       end
 
       def switch_host!(cname)
         Rails.application.routes.default_url_options[:host] = cname
         Hyrax::Engine.routes.default_url_options[:host] = cname
       end
+
+      def setup_tenant_cache
+        return unless Flipflop.enabled?(:cache_enabled).presence
+        Rails.application.config.action_controller.perform_caching = true
+        ActionController::Base.perform_caching = true
+        Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id }
+        Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
+      end
+
+      def disable_tenant_cache
+        Rails.application.config.action_controller.perform_caching = Settings.cache_enabled || false
```

Comment:
I missed this the first time i read the PR... the `|| false` at the end. 

What is `cache_enabled` returning unless its a boolean? If its not been set, then it would likely be `nil`, which is `falsey` and so equates to `false` when evaluated. 

If you desperately need the false there, which i cant see why you would, can you make it clearer what its doing by surrounding it with `()`, like: 

`Rails.application.config.action_controller.perform_caching = (Settings.cache_enabled || false)`

Now its super clear that its either of these values, depending on what is set.

Reply:
Should this be `Flipflop.enabled?(:cache_enabled)` not `Settings.cache_enabled`?

Reply:
I worked on this with @okeen but can't remember why we had to keep it like that

Reply:
There is nthing else in this PR that matches that, is it elsewhere?

Reply:
It was set in the docker-compose YAML but I am pretty sure it is resolving to to nil now

Reply:
Unless I was supposed to bring it over and didn't.............

Reply:
Yeah we had `SETTINGS__CACHE_ENABLED: "true"` in the docker-compose

Reply:
Is that going to be in the production YAML too? If so, you can get rid of the `|| false` as we control the default and it doesn't need to be cluttered up by code that will never be run.

Reply:
The reason why we use `Settings` when resetting the caching config is to set it by default to whatever the default application wide configuration is. This way you can enable the cache in a per tenant way and still have a default configuration to rely on for every tenant.

Also, consider that at that point, there is no active tenant, which means the FlipFlop features are not going to be loadable. We need to rely on Rails' env settings for that.

Reply:
Also, the reason why I wanted to ensure it was either `true` or `false` was to not mess up too much with ActionController, but I think you are right and it can safely be a nil.

Reply:
@okeen if you're using `present?` and `blank?` then `nil` is fine. I just wanted to check that it is actually a variable on the `Settings` object and isn't going to return a "method not found error".